control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller_test.go

```diff
@@ -1572,3 +1572,76 @@ func TestReconcileRouterServiceStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcileCoreIgnitionConfig(t *testing.T) {
+	awsHCP := &hyperv1.HostedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "aws-hcp",
+			Namespace: "clusters",
+		},
+		Spec: hyperv1.HostedControlPlaneSpec{
+			Platform: hyperv1.PlatformSpec{
+				Type: hyperv1.AWSPlatform,
+				AWS:  &hyperv1.AWSPlatformSpec{},
+			},
+		},
+	}
+
+	powervsHCP := &hyperv1.HostedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "powervs-hcp",
+			Namespace: "clusters",
+		},
+		Spec: hyperv1.HostedControlPlaneSpec{
+			Platform: hyperv1.PlatformSpec{
+				Type:    hyperv1.PowerVSPlatform,
+				PowerVS: &hyperv1.PowerVSPlatformSpec{},
+			},
+		},
+	}
+
+	tests := []struct {
+		name               string
+		hcp                *hyperv1.HostedControlPlane
+		expectedConfigMaps []client.Object
+	}{
+		{
+			name: "AWS platform with standard ignition config",
+			hcp:  awsHCP,
+			expectedConfigMaps: []client.Object{
+				&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: "clusters", Name: "ignition-config-fips"}},
+				&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: "clusters", Name: "ignition-config-worker-ssh"}},
+			},
+		},
+		{
+			name: "PowerVS platform with multipath and other standard ignition config",
+			hcp:  powervsHCP,
+			expectedConfigMaps: []client.Object{
```

Comment:
These look good, but could you please use `testutil.CompareWithFixture` and actually assert over the content in these? This test would pass if the ConfigMaps had no data :)

Reply:
Agree, added the testdata and updated the tests accordingly, please take a look now!